test/index.ts

```diff
@@ -0,0 +1,172 @@
+import {Calculator, Operation} from '../src/Calculator';
+import {Webpage} from '../src/Webpage';
+import * as ServerMock from 'mock-http-server';
+import {expect, assert} from 'chai';
+import * as sinon from 'sinon';
+
+describe('Level 2 - Unit Testing (Calculator)', () =>
+{
+    let calc: Calculator;
+
+    beforeEach(() => calc = new Calculator());
+
+    it('should return 5 when added 3 and 2', () =>
+    {
+        calc.input(3);
+        calc.operation(Operation.ADD);
+        calc.input(2);
+        
+        expect(calc.result).to.equal(5);
+    });
+
+    it('should return 1 when subtracted 3 and 2', () =>
+    {
+        calc.input(3);
+        calc.operation(Operation.SUB);
+        calc.input(2);
+        
+        expect(calc.result).to.equal(1);
+    });
+
+    it('should return 6 when multiplied 3 and 2', () =>
+    {
+        calc.input(2);
+        calc.operation(Operation.MUL);
+        calc.input(3);
+        
+        expect(calc.result).to.equal(6);
+    });
+
+    it('should return 1 when divided 3 and 3', () =>
+    {
+        calc.input(3);
+        calc.operation(Operation.DIV);
+        calc.input(3);
+        
+        expect(calc.result).to.equal(1);
+    });
+
+    it('should throw "Division by 0" error', () =>
+    {    
+        calc.input(3);
+        calc.operation(Operation.DIV);
+
+        expect(calc.input.bind(calc, 0)).to.throw('Cannot divide by 0. Insert a different value.');
+    });
+
+    it('should throw "Pending operation" error when is tried to get the result before the an operation executed', () =>
+    {    
+        calc.input(3);
+        calc.operation(Operation.SUB);
+
+        expect(() => calc.result).to.throw('Pending operation... Insert a value.');
+    });
+
+    it('should throw "Pending operation" error when is tried to change the operation type before the operation itself', () =>
+    {    
+        calc.input(3);
+        calc.operation(Operation.SUB);
+
+        expect(calc.operation.bind(calc, Operation.ADD)).to.throw('Pending operation... Insert a value.');
+    });
+
+    it('should throw "Unknown operation." error when is tried to set an uknown value as operation (not corresponding to any element decalred in the Operation enum)', () =>
+    {    
+        calc.operation(99);
+
+        expect(() => calc.input(3)).to.throw('Unknown operation.');
+    });
+});
+
+describe('Level 2 - Unit Testing (Webpage)', () => 
+{
+    const server = new ServerMock({ host: 'localhost', port: 9000 });
+    let web: Webpage;
+
+    beforeEach((done) => 
+    {
+        server.start(done);
+        web = new Webpage();
```

Comment:
Are you sure this works this way?
`server.start` call `done` when completed, so the test may start before the `web = new Webpage()` get called.
You should swap this two instruction.